lib/generator.js

```diff
@@ -123,6 +134,8 @@ class Generator {
     this.hooks = {};
     /** @type {Object} Maps schema URL to folder. */
     this.mapBaseUrlToFolder = mapBaseUrlToFolder;
+    /** @type {Object} Npm registry information. */
```

Comment:
for better jsdoc experience can you please:
- add above class type definition for the `registry` option like https://github.com/asyncapi/parser-js/blob/master/lib/parser.js#L46-L53
- If you call this type `RegistryOptions` then `/** @type {Object} Npm registry information. */` will have to be `/** @type {RegistryOptions} Npm registry information. */`. If I'm not mistaken as a result we will get a nice API docs generated
- I think that once you add `typedef` as described above, then you can also reuse it in jsdoc for the constructor, so you do not have to repeat all these options and descriptions as they are there now